code/clue.js

```diff
@@ -1,59 +1,495 @@
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-const weapons = []
+// Empty object to store the randomly picked killer, weapon and room.
+let mystery = {
+  killer: "",
+  weapon: "",
+  room: "",
+};
 
-const rooms = []
+// Randomly select an item from an array
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+// Fill an array with similar objects
+const fillArray = (item, amountOfTimes) => {
+  let result = [];
+  for (let i = 0; i < amountOfTimes; i++) {
+    result.push(item);
+  }
+  return result;
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// Create new weapon array if suspect has a favourite weapon
+const favouriteWeaponArray = (weapons) => {
+  if (mystery.killer.favouriteWeapon) {
+    const newWeaponArray = [
+      ...weapons,
+      ...fillArray(mystery.killer.favouriteWeapon, 5),
+    ];
+    return newWeaponArray;
+  } else {
+    return weapons;
+  }
+};
 
+// Create new room array if a weapon is picked
+const favouriteRoomArray = (rooms) => {
+  if (mystery.weapon) {
+    const onlyFavouriteRooms = fillArray(mystery.weapon, 8);
+    const newRoomArray = [...rooms];
+
+    onlyFavouriteRooms.forEach((item, index) => {
+      newRoomArray.push(onlyFavouriteRooms[index].resident);
+
+      index++;
+    });
+    return newRoomArray;
+  } else {
+    return rooms;
+  }
+};
+
+// Randomly select a favourite weapon from an array
+const shuffleFavouriteWeapon = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
+
+// Remove active from class if it already exists, add if not
+const toggleActive = (front, back) => {
+  front.classList.toggle("active");
+  back.classList.toggle("active");
+};
+
+// Show card back, hide card front
+const showCardBack = (back, front) => {
+  back.style.display = "flex";
+  front.style.display = "none";
+  front.style.opacity = 0;
+};
+
+// Make card front fade in, hide card back
+const showCardFront = (front, back) => {
+  front.style.display = "flex";
+  back.style.display = "none";
+
+  // Make card fade in
+  setTimeout(() => (front.style.opacity = 1), 50);
+};
+
+// Reset reveal button and reveal message from previous reveals
+const resetReveal = () => {
+  const revealButton = document.getElementById("revealButton");
+  const mysteryReveald = document.getElementById("mysteryReveald");
+  revealButton.classList.remove("reveald");
+  revealButton.innerText = "Reveal the crime";
+  mysteryReveald.innerHTML = "Who did it?";
```

Comment:
It's only text you are passing, so instead of `.innerHTML` you should use `innerText`